src/shell.js

```diff
@@ -96,7 +96,7 @@ var ENVIRONMENT_IS_WEB = typeof window == 'object';
 var ENVIRONMENT_IS_WORKER = typeof importScripts == 'function';
 // N.b. Electron.js environment is simultaneously a NODE-environment, but
 // also a web environment.
-var ENVIRONMENT_IS_NODE = typeof process == 'object' && typeof process.versions == 'object' && typeof process.versions.node == 'string';
+var ENVIRONMENT_IS_NODE = typeof process == 'object' && typeof process.versions == 'object' && typeof process.versions.node == 'string' && process.type != 'renderer';
```

Comment:
Does electron set `process.versions.node`?  It looks like the check for `process.versions.node` was added specifically to deal with electron.

Reply:
> Does electron set `process.versions.node`?

Yes
![image](https://github.com/user-attachments/assets/4b6d9264-ce3e-469e-bfb0-66ac201bf5ec)

Reply:
https://www.electronjs.org/docs/latest/tutorial/electron-timelines
Electron v22.0.0 comes with Node.js as v16.17.
Electron v22.0.0 was released in 2022, while https://github.com/emscripten-core/emscripten/commit/e9971d17e02d3fc3861f35d629a29baa6931842c is for 2019.

Reply:
> > Does electron set `process.versions.node`?
> 
> Yes ![image](https://private-user-images.githubusercontent.com/96647698/370846633-4b6d9264-ce3e-469e-bfb0-66ac201bf5ec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyOTE0MjQsIm5iZiI6MTcyNzI5MTEyNCwicGF0aCI6Ii85NjY0NzY5OC8zNzA4NDY2MzMtNGI2ZDkyNjQtY2UzZS00NjllLWJmYjAtNjZhYzIwMWJmNWVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI1VDE5MDUyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmNTEzMDE4MzQzNzgyNWFiYzcyM2Q1Yjk1YzMzZGIwYTE0ODE4MGY0MDJhMTkzYjY0MWRiMDY1YmJhZWYzY2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1bfYeb8aijzTKRcw6nv7FTK1SX-kKuy2TySfwRf-IDs)

`process` from `contextBridge.exposeInMainWorld('process', process)` in the preload script.

Reply:
So its possible to expose the `process` module in some way.. but `process.type` will always be `renderer`?

In that case this LGTM, but can you drop the `process.versions.node` check?

Again I wish we had some kind of testing for electron in our CI :(

Reply:
> So its possible to expose the `process` module in some way.. but `process.type` will always be `renderer`?

Yes
![image](https://github.com/user-attachments/assets/8090b9cc-94d9-4ab7-af3c-9b46b259aa7f)
This is the result under Electron v32

Reply:
> In that case this LGTM, but can you drop the `process.versions.node` check?

Feels like it's possible, but not sure it'll work on a version that's too old.

Reply:
Actually I think we do still want the `process.versions.node` check, so that if `process` is defined by some other means its doesn't trigger this to be true.